app/models/post_action_type.rb

```diff
@@ -25,62 +42,69 @@ def replace_flag_settings(settings)
 
     def types
       if overridden_by_plugin_or_skipped_db?
-        return Enum.new(like: 2).merge!(flag_settings.flag_types)
+        return Enum.new(like: LIKE_POST_ACTION_ID).merge!(flag_settings.flag_types)
       end
-      Enum.new(like: 2).merge(flag_types)
+      Enum.new(like: LIKE_POST_ACTION_ID).merge(flag_types)
     end
 
     def expire_cache
       Discourse.redis.keys("post_action_types_*").each { |key| Discourse.redis.del(key) }
       Discourse.redis.keys("post_action_flag_types_*").each { |key| Discourse.redis.del(key) }
+      Discourse.cache.delete(POST_ACTION_TYPE_ALL_FLAGS_KEY)
+      Discourse.cache.delete(POST_ACTION_TYPE_PUBLIC_TYPE_IDS_KEY)
     end
 
     def reload_types
       @flag_settings = FlagSettings.new
-      ReviewableScore.reload_types
       PostActionType.new.expire_cache
+      PostActionType.expire_cache
+      ReviewableScore.reload_types
     end
 
     def overridden_by_plugin_or_skipped_db?
       flag_settings.flag_types.present? || GlobalSetting.skip_db?
     end
 
     def all_flags
-      Flag.unscoped.order(:position).all
+      Discourse
+        .cache
+        .fetch(PostActionType::POST_ACTION_TYPE_ALL_FLAGS_KEY) do
+          Flag
+            .unscoped
```

Comment:
Just want to confirm we need this because I don't see a `default_scope` being set anywhere.

Reply:
It is here 

https://github.com/discourse/discourse/blob/main/app/models/flag.rb#L19